.github/workflows/nightly-build.yml

```diff
@@ -48,25 +48,39 @@ jobs:
     - name: Build with Maven
       run: mvn -B clean install -U -Pintegration
     - name: Build docker images
+      id: build
       run:
-          ./dockerfiles/build.sh
+          SHORT_SHA1=$(git rev-parse --short HEAD)
+          echo ::set-output name=short_sha1::${SHORT_SHA1}
+          ./dockerfiles/build.sh --tag:nightly --sha-tag
     - name: Push docker images
       run: |
         docker push quay.io/eclipse/che-endpoint-watcher:nightly
+        docker push quay.io/eclipse/che-endpoint-watcher:${{ steps.build.outputs.short_sha1 }}
+
```

Comment:
why build both nightly and next in parallel? seems like a waste of resources and duplication of tags.